### PR TITLE
Adds attempt at implementing "open" functionality to demonstrate which cells are already open 

### PR DESCRIPTION
**Pull request by bahmady**
Hi @arnau, I've no idea if this is a "good" approach but it's an approach which "works", and I'm following the idea that "it does what I want it to do so it's a good starting point". 

I don't necessarily think this should be merged as-is into `master` but I'd like to use this PR to demonstrate what I did do (and which "works") as a means for discussion. Ideally I'd like it if at least some of this were sensible but we'll see :D 

If I am to carry on down this train of thought, I would introduce a way for the user to interact with the game. As it is I'm using `playground.rb` to prove in a very basic way that the "popping" works and that the instance is still modified ("popped") within the execution of `playground.rb`. 

Sorry it took so long to get this out!

### CODE REVIEW

**Comment by bahmady** [comment]
@arnau I should really separate this into a separate PR but I've added a first solution to today's problem in `test.rb`. It returns:

X,X,X,X
X,X,O,X
X,O,X,O
X,O,O,O

**Comment by bahmady** [comment]
I've also added a unit test for opened cells

File: lib/minesweeper/grid.rb [hunk 1]
```diff
@@ -33,15 +33,28 @@ def neighbours(x, y)
     end
 
     def unseal(x, y)
-      if cell(x, y).mine?
-        false
-      else
-        neighbours(x, y).count { |n| n.mine? }
+      if cell(x, y).mine? == false and cell(x, y).open? == false
```
Comment (arnau) [inline lib/minesweeper/grid.rb]: Remember that in ruby you don't need to compare boolean values, it's redundant. This is better:

```ruby
if not cell(x, y).mine? and not cell(x, y).open?
```

Which can be changed to:

```ruby
if not (cell(x, y).mine? or cell(x, y).open?)
```

Which can be changed to:

```ruby
unless cell(x, y).mine? or cell(x, y).open?
```

File: lib/minesweeper/grid.rb [hunk 2]
```diff
@@ -33,15 +33,28 @@ def neighbours(x, y)
...
+        cell(x,y).open
+        num_mines = neighbours(x, y).count { |n| n.mine? }
+        return "There are #{num_mines} mines around that cell and #{num_open_cells} open cells"
```
Comment (arnau) [inline lib/minesweeper/grid.rb]: `return` is not needed. Ruby implicitly returns the latest expression.

Also, you should not return a text here. You should be careful mixing things that you think are the way to go and things that are experiments. If you keep mixing them up you'll end up with a mess to untangle later on. Not fun, I can guarantee that.

File: lib/minesweeper/grid.rb [hunk 3]
```diff
@@ -33,15 +33,28 @@ def neighbours(x, y)
...
+    def xdimension
+      @grid.count
+    end
+
+    def ydimension
+      @grid[0].count
+    end
+
+    def num_cells
+      xdimension*ydimension
```
Comment (arnau) [inline lib/minesweeper/grid.rb]: Spaces matter. `xdimension * ydimension`.

File: spec/cell_spec.rb [hunk 4]
```diff
@@ -7,4 +7,9 @@
   end
 end
 
-
+describe Minesweeper::Cell do
+  it "should produce mines which can be open or not" do
+    actual = Minesweeper::Cell.new().open?
+    expect(actual).to be(true).or be(false)
```
Comment (arnau) [inline spec/cell_spec.rb]: This is true but not quite true right? Every time you create a cell, no matter what, you expect it to be closed.

Once you open it, you expect the cell to be open, always. Right?

File: test.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  end
+  row[3..3].each do |cell|
+    if cell == 1
+      print "X"
+    else
+      print "O"
+    end
+  end
+  print "\n"
+end
```
Comment (arnau) [inline test.rb]: This is a solution but not a good one. If I change the dimensions of the array your code will do something unexpected.

File: spec/cell_spec.rb (hunk 4 above)
Comment (bahmady) [reply spec/cell_spec.rb]: 🤔  yes, I'll rethink this.

File: test.rb (hunk 5 above)
Comment (bahmady) [reply test.rb]: 💩 yeah I'm struggling with this; I just wanted to get "a solution" out there and then iterate from there, but the more I mess around with this one the more I think I'm better off doing something else entirely.

File: lib/minesweeper/grid.rb (hunk 2 above)
Comment (bahmady) [reply lib/minesweeper/grid.rb]: OK, I've removed it :D

**Comment by bahmady** [comment]
@arnau this PR is a bit out of control now but I've made some changes and everything seems to work now, including the new tests

**Comment by bahmady** [comment]
Cool! I'll have a look today separately :D

**Pull request merged**